dirty_cat/gamma_poisson_factorization.py

```diff
@@ -0,0 +1,543 @@
+    3. Assuming that n-grams counts follow a Poisson law, we fit H and W to
+       maximize the likelihood of the data, with a Gamma prior for the
+       activations H to induce sparsity.
+    4. In practice, this is equivalent to a non-negative matrix factorization
+       with the Kullback-Leibler divergence as loss, and a Gamma prior on H.
+       We thus optimize H and W with the multiplicative update method.
+"""
+import numpy as np
+from scipy import sparse
+from sklearn.utils import check_random_state, gen_batches
+from sklearn.utils.extmath import row_norms, safe_sparse_dot
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.feature_extraction.text import CountVectorizer, HashingVectorizer
+from sklearn.cluster import KMeans
+from sklearn.neighbors import NearestNeighbors
+from sklearn.cluster.k_means_ import _k_init
+from sklearn.decomposition.nmf import _beta_divergence
+
+
+class OnlineGammaPoissonFactorization(BaseEstimator, TransformerMixin):
+    """
+    Online Gamma-Poisson Factorization by minimizing the
+    Kullback-Leibler divergence.
+
+    Parameters
+    ----------
+
+    n_topics : int, default=10
+        Number of topics of the matrix factorization.
+
+    batch_size : int, default=512
+        Number of samples per batch.
+
+    gamma_shape_prior : float, default=1.1
+        Shape parameter for the Gamma prior distribution.
+
+    gamma_scale_prior : float, default=1.0
+        Scale parameter for the Gamma prior distribution.
+
+    rho : float, default=0.95
+        Weight parameter for the update of the W matrix.
+    
+    rescale_rho : boolean, default=False
+        If true, use rho ** (batch_size / len(X)) instead of rho to obtain an
+        update rate per iteration that is independent of the batch size.
+
+    hashing : boolean, default=False
+        If true, HashingVectorizer is used instead of CountVectorizer.
+
+    hashing_n_features : int, default=2**12
+        Number of features for the HashingVectorizer. Only relevant if
+        hashing=True.
+
+    init : str, default='k-means++'
+        Initialization method of the W matrix.
+        Options: {'k-means++', 'random', 'k-means'}.
+        If init='k-means++', we use the init method of sklearn.cluster.KMeans.
+        If init='random', topics are initialized with a Gamma distribution.
+        If init='k-means', topics are initialized with a KMeans on the n-grams
+        counts. This usually makes convergence faster but is a bit slower.
+
+    tol : float, default=1e-4
+        Tolerance for the convergence of the matrix W.
+
+    min_iter : int, default=2
+        Minimum number of iterations on the input data.
+
+    max_iter : int, default=5
+        Maximum number of iterations on the input data.
+
+    ngram_range : tuple, default=(2, 4)
+        The range of ngram length that will be used to build the
+        bag-of-n-grams representation of the input data.
+
+    analizer : str, default='char'.
```

Comment:
Wouldn't it be ``analyzer`` instead of ``analizer`` ?

Reply:
You're right, I'll fix this !